src/packet_distribution.cpp

```diff
@@ -722,6 +722,36 @@ void AOApplication::server_packet_received(AOPacket *p_packet)
                                f_contents.at(4) == "1",
                                f_contents.at(5) == "1");
   }
+  else if (header == "TI") { // Timer packet
+    if (courtroom_constructed && f_contents.size() > 0) {
+      qint64 resolution = f_contents.at(0).toInt();
+      //Type 0 = start/stop timer
+      //Type 1 = pause timer
+      int type = 0;
+      if (f_contents.size() > 1)
+        type = f_contents.at(1).toInt();
+      qDebug() << "timer" << resolution << last_ping << resolution - last_ping;
+      resolution = resolution - last_ping;
+      if (resolution > 0)
+      {
+        if (type == 1)
+        {
+          w_courtroom->pause_clock();
+          w_courtroom->set_clock(resolution);
+        }
+        else
+          w_courtroom->start_clock(resolution);
+      }
+      else
+        w_courtroom->stop_clock();
+    }
+  }
+  else if (header == "CHECK") {
+    if (courtroom_constructed) {
+      last_ping = w_courtroom->get_ping();
+      w_courtroom->set_window_title(window_title + " [ping:" + QString::number(last_ping) + "]");
```

Comment:
imo this looks kind of ugly. also, there should be an option to toggle this.

Reply:
I dunno how else to display it tbh, any ideas?

Reply:
You typically either add it to some kind of player list, or add it as a clientside slash command. But since there is no player list and you don't want any new clientside commands, then it may have to stay hidden.